### ISSUE TITLE
Is it the project still alive? Django 3 support?

### ISSUE THREAD

**Issue by Cicciodev** [fusionbox/django-betterforms#58]
Since the project use python_2_unicode_compatible decorator and django3 dropped support for python2 maybe it's time to remove it or if the aim is to make it still compatible with older project maybe the solution could be something like create a doing-nothing decorator if import fails:
```
try:
    from django.utils.encoding import python_2_unicode_compatible
except ImportError:
    def python_2_unicode_compatible(klass):
        """
        Since python2 no more supported in django3 
        it would suffice to create a fake decorator if import fails
        """
        return klass
```

**Comment by jpic** [comment]
Not sure if it's really worth maintaining backward compatibility: the releases for Python 2 work great, upstream is stable and in the worst case there will be a minor patch release once a year until Red Hat completely drops Python 3. 

What do you think ?

**Comment by Cicciodev** [comment]
I'm not a fan of backward compatibility, I proposed it beacuse I've seen the project is not being updated from long. I supposed that maybe it is widely used by "ancient" project.

I don't think that my opinion count much but I've seen that you have made a pull request and I'm really glad!

**Comment by bluesurfer** [comment]
Any news on this?

**Comment by Cicciodev** [comment]
@bluesurfer a pull request has been created by @jpic 
Sorry but I'm not sure how pull request in public project works so I don't know when it will be closed!

**Comment by jpic** [comment]
Yep, meanwhile I use a fork package that includes the said patches, you can use it to if you want  it's django-betterforms-jpic

**Comment by Asday** [comment]
@jpic You might wanna update the details in your fork - PyPI links to the wrong repository, has the wrong install instructions in the readme, lists the wrong Django versions as supported, etc.

**Comment by jpic** [comment]
@Asday it's just meant as a temporary solution, perhaps the repo owners should transfer the repo to JazzBand and move on to whatever they are maintaining now.

**Comment by verngutz** [comment]
How to use the temporary fork? I tried uninstalling the original and installing the fork. I then get `ModuleNotFoundError: No module named 'betterforms'` if I put `betterforms` in `INSTALLED_APPS`. It used to work fine without this setting, using the original project, so I tried removing it, but then neither `import betterforms.form` nor `import betterforms.multiform` work.

**Comment by Asday** [comment]
@verngutz Worth opening an issue on @jpic's repo, but personally I didn't have that issue at all.  When you make the issue, make sure you include your `requirements.txt` and all relevant code snippets.

**Comment by jpic** [comment]
Honnestly I just changed the requirements from django-betterforms to https://github.com/jpic/django-betterforms/tarball/master and without changing any code it works for me.

Does it work in a new virtual environment ?

**Comment by verngutz** [comment]
Welp. Silly me. I installed @jpic's fork first before uninstalling the original one. So of course, since they both went into the same folder in site-packages, the uninstallation removed everything. The egg-info remains, telling pip that the requirement is still satisfied even though it isn't, so this took me a while to figure out.

**Comment by Asday** [comment]
Better to require `django-betterforms-jpic` in your requirements.  It has the same named module so no code changes are required, but it's a different package, so `pip` does the right thing.

**Comment by corinnamk** [comment]
I uninstalled django-betterforms and installed django-betterforms-jpic instead. Now I get the following error: `pkg_resources.DistributionNotFound: The 'django-betterforms' distribution was not found and is required by the application.` 

Any ideas how to fix this? 

(or: has anyone found a better solution how to use class based views for multiple models and forms?)

**Comment by jpic** [comment]
The -jpic version works if you install it with git, I've removed
django-betterforms dependency from my project (crudlfap) so I won't be
doing nor expecting anything else on this, sorry.

**Comment by aimestereo** [comment]
> The -jpic version works if you install it with git, I've removed
django-betterforms dependency from my project (crudlfap) so I won't be
doing nor expecting anything else on this, sorry.

exactly - it's the best solution

@corinnamk 

> I uninstalled django-betterforms and installed django-betterforms-jpic instead. Now I get the following error: `pkg_resources.DistributionNotFound: The 'django-betterforms' distribution was not found and is required by the application.`
>
> Any ideas how to fix this?

use the solution from @jpic - install it from git
Answering **why** this happens: @jpic by providing alternative pip package made an error, his package named `django-betterforms-jpic`, but in code he left the old name, so package resolver can't find it.

https://github.com/jpic/django-betterforms/blob/40da7eee2a1b1d9d026a16dbe56b1270c0f859cc/betterforms/__init__.py#L1

```
__version__ = __import__('pkg_resources').get_distribution('django-betterforms').version
```

**Comment by BismeetSingh** [comment]
How do I install jpic from git?

**Comment by Asday** [comment]
@BismeetSingh I personally have the following in my `requirements.txt`:

```
django-betterforms-jpic==1.2.2
```

**Comment by CSprat451** [comment]
We apologize for letting this project stagnate for so long. We will be releasing version 2.0 in the near future, which supports Django 3.x and 4.0 and we hope to maintain it more actively going forward.
